WordPress/WordPressShare/ShareViewController.swift

```diff
@@ -0,0 +1,156 @@
+        }
+    }
+
+    override func configurationItems() -> [AnyObject]! {
+        let blogPickerItem = SLComposeSheetConfigurationItem()
+        blogPickerItem.title = NSLocalizedString("Post to:", comment: "Upload post to the selected Site")
+        blogPickerItem.value = selectedSiteName ?? NSLocalizedString("Select a site", comment: "Select a site in the share extension")
+        blogPickerItem.tapHandler = { [weak self] in
+            self?.displayBlogPicker()
+        }
+        
+        let statusPickerItem = SLComposeSheetConfigurationItem()
+        statusPickerItem.title = NSLocalizedString("Post Status:", comment: "Post status picker title in Share Extension")
+        statusPickerItem.value = self.postStatuses[postStatus]!
+        statusPickerItem.tapHandler = { [weak self] in
+            self?.displayStatusPicker()
+        }
+        
+        return [blogPickerItem, statusPickerItem]
+    }
+
+    
+    private func displayBlogPicker() {
+        let pickerViewController = BlogPickerViewController()
+        pickerViewController.onChange = { (siteId, description) in
+            self.selectedSiteID = siteId
+            self.selectedSiteName = description
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    private func displayStatusPicker() {
+        let pickerViewController = PostStatusPickerViewController()
+        pickerViewController.statuses = postStatuses
+        pickerViewController.onChange = { (status, description) in
+            self.postStatus = status
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    
+    // TODO: This should eventually be moved into WordPressComKit
+    private let postStatuses = [
+        "draft" : NSLocalizedString("Draft", comment: "Draft post status"),
+        "publish" : NSLocalizedString("Publish", comment: "Publish post status")]
+
+    private func splitContentTextIntoSubjectAndBody(contentText: String) -> (subject: String, body: String) {
+        let fullText = contentText
+        let firstCarriageReturnIndex = fullText.rangeOfCharacterFromSet(NSCharacterSet.newlineCharacterSet())
+        let firstLineOfText = firstCarriageReturnIndex != nil ? fullText.substringToIndex(firstCarriageReturnIndex!.startIndex) : fullText
+        let restOfText = firstCarriageReturnIndex != nil ? fullText.substringFromIndex(firstCarriageReturnIndex!.endIndex) : ""
+
+        return (firstLineOfText, restOfText)
+    }
+    
+    private func contentWithSourceURL(url: NSURL?) -> String {
+        guard let url = url else {
+            return contentText
+        }
+        
+        // Append the URL to the content itself
+        return contentText + "\n\n<a href=\"\(url.absoluteString)\">\(url.absoluteString)</a>"
```

Comment:
@koke `contentText` and `attributedContentText` isn't *always* the title. If you select a snippet of text from the website, both strings will actually contain the "Highlighted Text", so i'm afraid the solution isn't 100% reliable. 

[Screenshot here](https://cloudup.com/c3KhDchWjIF), tested [with this page](http://www.lantean.co/arc-weakself-caveats/).